sigoptlite/sigoptlitetest/endpoints/test_best_assignments.py

```diff
@@ -128,107 +160,155 @@ def test_multimetric_thresholds(self, threshold_y1, threshold_y2):
       ],
       observation_budget=123,
     )
-    experiment = LocalExperimentBuilder(experiment_meta)
-    pareto_optimal_observations = self.create_observations_with_values_in_simplex(experiment, on_boundary=True)
-    non_optimal_observations = self.create_observations_with_values_in_simplex(experiment, on_boundary=False)
-    observations = pareto_optimal_observations + non_optimal_observations
-    random.shuffle(observations)
+    e = self.conn.experiments().create(**experiment_meta)
+
+    num_optimal = 15
+    pareto_optimal_assignments = self.generate_random_assignments(experiment_meta, num=num_optimal)
+    non_optimal_assignments = self.generate_random_assignments(experiment_meta)
+    assignments_list = pareto_optimal_assignments + non_optimal_assignments
+
+    pareto_optimal_values = self.create_values_in_simplex(on_boundary=True, num=num_optimal)
+    non_optimal_values = self.create_values_in_simplex(on_boundary=False)
+    values_list = pareto_optimal_values + non_optimal_values
+
+    self.create_shuffled_observations(self.conn, assignments_list, values_list)
+    best_assignments_list = list(self.conn.experiments(e.id).best_assignments().fetch().iterate_pages())
+    self.assert_best_assignments_are_sorted(e, best_assignments_list)
 
     if threshold_y1 is None:
       threshold_y1 = 0
-
     if threshold_y2 is None:
       threshold_y2 = 0
-
-    best_assignments_logger = BestAssignmentsLogger(experiment)
-    best_observations_from_logger = best_assignments_logger.fetch(observations)
-    self.assert_observations_are_sorted(best_observations_from_logger, experiment)
-
-    expected_pareto_optimal_observations = [o.get_client_observation(experiment) for o in pareto_optimal_observations]
-    for observation in expected_pareto_optimal_observations:
-      if observation["values"][0]["value"] >= threshold_y1 and observation["values"][1]["value"] <= threshold_y2:
-        assert observation in best_observations_from_logger
-      else:
-        assert observation not in best_observations_from_logger
+    for best_assignment in best_assignments_list:
```

Comment:
can also `assert len(best_assignments_list) < num_optimal`, since we are thresholding a portion of the pareto frontier.